services/wallet/transfer/commands_sequential.go

```diff
@@ -730,17 +775,53 @@ type loadBlocksAndTransfersCommand struct {
 
 	// Not to be set by the caller
 	transfersLoaded map[common.Address]bool // For event RecentHistoryReady to be sent only once per account during app lifetime
+	loops           int
+	errorCounter
+	mu sync.Mutex
 }
 
-func (c *loadBlocksAndTransfersCommand) Run(parent context.Context) error {
-	log.Debug("start load all transfers command", "chain", c.chainClient.NetworkID(), "accounts", c.accounts)
+func (c *loadBlocksAndTransfersCommand) incStarted() {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.loops++
```

Comment:
matter of preference, if it's just to have a simple thread-safe counter or boolean flag I like using atomics instead, which normally have less overhead